lib/chef/mixin/shell_out.rb

```diff
@@ -153,15 +153,23 @@ def self.clean_array(*args)
         args.flatten.compact.map(&:to_s)
       end
 
+      def self.transport_connection
+        Chef.run_context.transport_connection
+      end
+
       def self.shell_out_command(*args, **options)
-        cmd = if options.empty?
-                Mixlib::ShellOut.new(*args)
-              else
-                Mixlib::ShellOut.new(*args, **options)
-              end
-        cmd.live_stream ||= io_for_live_stream
-        cmd.run_command
-        cmd
+        if Chef::Config.target_mode?
+          FakeShellOut.new(args, options, transport_connection.run_command(args.join(" "))) # FIXME FIXME FIXME: join here is horrible
+        else
+          cmd = if options.empty?
+                  Mixlib::ShellOut.new(*args)
+                else
+                  Mixlib::ShellOut.new(*args, **options)
```

Comment:
I don't think you need to do the conditional here. `**empty_hash` should be the same as nothing at all.